cmd2/cmd2.py

```diff
@@ -2977,7 +2977,7 @@ def do_set(self, args: argparse.Namespace) -> None:
         if orig_value != new_value:
             onchange_hook = getattr(self, '_onchange_{}'.format(param), None)
             if onchange_hook is not None:
-                onchange_hook(old=orig_value, new=new_value)
+                onchange_hook(old=orig_value, new=new_value)  # pylint: disable=not-callable
```

Comment:
Would it be better to do something like the following:
```Python
if onchange_hook is not None and callable(onchange_hook):
    onchange_hook(old=orig_value, new=new_value)
```

Reply:
I dunno. If there is an attribute that a developer created called `_onchange_mysetting` that isn't a callable, then the code as written would generate an exception. Same thing if it was callable, but took 1 (or 3) parameters. Is it better to silently ignore the malformed or uncallable onchange hook, or to throw an exception?

By way of reference, in the new implementation of hooks, it checks both callable and method signature when you register a hook, and won't register the hook if the registered callable doesn't have the right number of parameters that are annotated with the expected types for each parameter. The registration hook method throws TypeErrors if anything is amiss.

I was just trying to get rid of the pylint error. Pylint still generates the error even if we check if the hook is callable.

I've pushed a new commit with the proposed change, because even it could still throw errors if the method has the wrong number of parameters, at least we won't try something that isn't callable.